test/e2e/install/install_test.go

```diff
@@ -150,13 +146,41 @@ func mayastorReadyPodCount() int {
 	return int(mayastorDaemonSet.Status.NumberAvailable)
 }
 
-func moacReadyPodCount() int {
+func moacReady() bool {
 	var moacDeployment appsV1.Deployment
 	if k8sClient.Get(context.TODO(), types.NamespacedName{Name: "moac", Namespace: "mayastor"}, &moacDeployment) != nil {
-		fmt.Println("Failed to get MOAC deployment")
-		return -1
+		logf.Log.Info("Failed to get MOAC deployment")
+		return false
 	}
-	return int(moacDeployment.Status.AvailableReplicas)
+
+	// { Remove/Reduce verbosity once we have fixed install test occasional failure.
+	logf.Log.Info("moacDeployment.Status",
+		"ObservedGeneration", moacDeployment.Status.ObservedGeneration,
+		"Replicas", moacDeployment.Status.Replicas,
+		"UpdatedReplicas", moacDeployment.Status.UpdatedReplicas,
+		"ReadyReplicas", moacDeployment.Status.ReadyReplicas,
+		"AvailableReplicas", moacDeployment.Status.AvailableReplicas,
+		"UnavailableReplicas", moacDeployment.Status.UnavailableReplicas,
+		"CollisionCount", moacDeployment.Status.CollisionCount)
+	for ix, condition := range moacDeployment.Status.Conditions {
+		logf.Log.Info("Condition", "ix", ix,
+			"Status", condition.Status,
+			"Type", condition.Type,
+			"Message", condition.Message,
+			"Reason", condition.Reason)
+	}
+	// }
+
+	for _, condition := range moacDeployment.Status.Conditions {
+		if condition.Type == appsV1.DeploymentAvailable {
+			if condition.Status == coreV1.ConditionTrue {
+				logf.Log.Info("MOAC is Available")
+				return true
+			}
+		}
+	}
+	logf.Log.Info("MOAC is Not Available")
```

Comment:
```suggestion
	logf.Log.Error("MOAC is Not Available")
```
My understanding is test cannot continue (otherwise I'd choose Warning).

Reply:
Log.Error has a different signature so
```
	err := errors.New("MOAC is not Available")
	logf.Log.Error(err, "")
```

Reply:
Doh! using logf.Log.Error is not correct as gingko triggers treats that as an assert.